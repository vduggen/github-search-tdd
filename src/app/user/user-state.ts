import IUser from './IUser';

interface IUserStoreState {
  user: IUser
}

const UserStoreState: IUserStoreState = {
  user: {
    name: '',
  },
};

export {
  UserStoreState,
  IUserStoreState,
};

export default {
  state: UserStoreState,
  mutations: {},
  actions: {},
  getters: {},
};
