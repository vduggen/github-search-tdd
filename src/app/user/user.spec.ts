// eslint-disable-next-line import/no-extraneous-dependencies
import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';

import userFixture from '@/tests/unit/fixtures/user';
import UserView from './user.vue';
import UserForm from './user-form/user-form.vue';
import UserProfile from './user-profile/user-profile.vue';

import { IUserStoreState, UserStoreState } from './user-state';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('UserView', () => {
  let state: IUserStoreState;

  function build() {
    const wrapper = shallowMount(UserView, {
      localVue,
      store: new Vuex.Store({ state }),
    });
    const findComponentUserForm = () => wrapper.findComponent(UserForm);
    const findComponentUserProfile = () => wrapper.findComponent(UserProfile);

    return {
      wrapper,
      userForm: findComponentUserForm,
      userProfile: findComponentUserProfile,
    };
  }

  beforeEach(() => {
    state = { ...UserStoreState };
  });

  it('renders the component', () => {
    const { wrapper } = build();

    const htmlWrapper = wrapper.html();
    expect(htmlWrapper).toMatchSnapshot();
  });

  it('renders main child components', () => {
    const { userForm, userProfile } = build();

    const existUserForm = userForm().exists();
    const existUserProfile = userProfile().exists();

    expect(existUserForm).toBe(true);
    expect(existUserProfile).toBe(true);
  });

  it('passes a binded user prop to user profile component', () => {
    state.user = userFixture;

    const { userProfile } = build();

    const userProfileVm = userProfile().vm as unknown as IUserStoreState;

    expect(userProfileVm.user).toBe(state.user);
  });
});
