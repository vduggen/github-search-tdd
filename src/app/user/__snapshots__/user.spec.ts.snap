// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div>
  <userform-stub></userform-stub>
  <userprofile-stub user="[object Object]"></userprofile-stub>
</div>
`;
